test/controlplane/suite/agent.go

```diff
@@ -74,6 +75,11 @@ func (h *agentHandle) setupCiliumAgentHive(clientset k8sClient.Clientset, extraC
 			func() *option.DaemonConfig { return option.Config },
 			func() cnicell.CNIConfigManager { return &fakecni.FakeCNIConfigManager{} },
 			func() gc.Enabler { return gc.NewFake() },
+			func() promise.Promise[resource.CRDSync] {
```

Comment:
Add a e.g. `resource.CRDSyncRejectedPromise` variable and use it here instead of repeating the code in tests.

Reply:
Done.